tests/Serializer/Doctrine/ObjectConstructorTest.php

```diff
@@ -295,6 +337,25 @@ private function createSerializerWithDoctrineObjectConstructor()
             ->addDefaultHandlers()
             ->build();
     }
+
+    /**
+     * @param string $className
+     * @return ClassMetadata
+     * @throws \Doctrine\Common\Annotations\AnnotationException
+     * @throws \JMS\Serializer\Exception\InvalidMetadataException
+     * @throws \ReflectionException
+     */
+    private function createClassMetadataFor($className): ClassMetadata
```

Comment:
you sure this is necessary? the default serializer metadata loader should fallback to annotations anyway

Reply:
As far as i can tell yes. the Constructor for ClassMetadata does not initialize anything by himself.
The old Code
```
$class = new ClassMetadata(Author::class);
```
Will result in an "empty" Class Metadata. For example the propertyMetadata array will simply be empty.
We need to initialize the Metadata since it is used for the extraction for the Identifier Field.

Is there a better way to Create a fully initialized ClassMetadata?